client/header/activity-panel/panels/orders.js

```diff
@@ -265,29 +274,37 @@ export default compose(
 			return { orders: [], isError: true, isRequesting: false, orderStatuses };
 		}
 
-		if ( hasActionableOrders ) {
+		// Query the core Orders endpoint for the most up-to-date statuses.
+		const allOrdersQuery = {
```

Comment:
Out of scope for this PR, but would be awesome if we could just return the IDs here - https://developer.wordpress.org/reference/classes/wp_query/#return-fields-parameter

Reply:
Done in 7a3f4fd.